src/main.rs

```diff
@@ -80,13 +113,28 @@ fn main() -> Result<i32> {
     };
 
     let mut client = Client::builder()
+        .use_rustls_tls()
         .http2_adaptive_window(true)
         .timeout(timeout)
         .redirect(redirect);
 
+    #[cfg(feature = "native-tls")]
+    if args.native_tls {
+        client = client.use_native_tls();
+    } else if utils::requires_native_tls(&args.url) {
+        // We should be loud about this to prevent confusion
+        warn("rustls does not support HTTPS for IP addresses. native-tls will be enabled. Use --native-tls to silence this warning.");
+        client = client.use_native_tls();
+    }
+
+    #[cfg(not(feature = "native-tls"))]
+    if args.native_tls {
+        return Err(anyhow!("This binary was built without native-tls support"));
```

Comment:
Although we don't have to use it, Anyhow has the `bail!("error")` macro as a shorthand for `return Err(anyhow!("error"))`.

Reply:
I think I'd rather not hide the control flow like that.